packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -216,167 +260,338 @@ export class EditableGrid extends ReactN.PureComponent<EditableGridProps, Editab
+        const hasSelection = editorModel.hasSelection();
+        let selectedColIdx = colIdx;
+        let selectedRowIdx = rowIdx;
+
+        switch (selection) {
+            case SELECTION_TYPES.ALL:
+                for (let c = 0; c < colCount; c++) {
+                    for (let r = 0; r < rowCount; r++) {
+                        selectionCells = selectionCells.add(genCellKey(c, r));
+                    }
+                }
+                break;
+            case SELECTION_TYPES.AREA:
+                selectedColIdx = editorModel.selectedColIdx;
+                selectedRowIdx = editorModel.selectedRowIdx;
+
+                if (hasSelection) {
+                    const upperLeft = [Math.min(selectedColIdx, colIdx), Math.min(selectedRowIdx, rowIdx)];
+                    const bottomRight = [Math.max(selectedColIdx, colIdx), Math.max(selectedRowIdx, rowIdx)];
+                    const maxColumn = Math.min(bottomRight[0], colCount - 1);
+                    const maxRow = Math.min(bottomRight[1], rowCount - 1);
+
+                    for (let c = upperLeft[0]; c <= maxColumn; c++) {
+                        for (let r = upperLeft[1]; r <= maxRow; r++) {
+                            selectionCells = selectionCells.add(genCellKey(c, r));
+                        }
+                    }
+                }
+                break;
+            case SELECTION_TYPES.SINGLE:
+                selectionCells = editorModel.selectionCells.add(genCellKey(colIdx, rowIdx));
+                break;
         }
-        const editorModel = this.getEditorModel();
-        if (editorModel.rowCount === 0 && this.props.initialEmptyRowCount > 0) {
-            addRows(this.getModel(this.props), this.props.initialEmptyRowCount);
+
+        if (selectionCells.size > 0) {
+            // if a cell was previously selected and there are remaining selectionCells then mark the previously
+            // selected cell as in "selection"
+            if (hasSelection) {
+                selectionCells = selectionCells.add(genCellKey(editorModel.selectedColIdx, editorModel.selectedRowIdx));
+            }
         }
+
+        return { selectedColIdx, selectedRowIdx, selectionCells };
     };
 
-    componentDidMount() {
-        document.addEventListener('click', this.onDocumentClick);
-        document.addEventListener('copy', this.onCopy);
-        document.addEventListener('paste', this.onPaste);
-    }
+    selectCell = (colIdx: number, rowIdx: number, selection?: SELECTION_TYPES, resetValue?: boolean): void => {
+        const { editorModel, onChange } = this.props;
+        const { cellValues, colCount, focusValue, rowCount } = editorModel;
 
-    componentWillUnmount() {
-        document.removeEventListener('click', this.onDocumentClick);
-        document.removeEventListener('copy', this.onCopy);
-        document.removeEventListener('paste', this.onPaste);
-    }
+        if (colIdx < 0 || rowIdx < 0 || colIdx > colCount) {
```

Comment:
should the last part be `colIdx >= colCount`?

Reply:
Probably, this was copied from actions.ts. I'll fix it, but I also think it's a non-issue because you can't actually trigger this event with incorrect parameters unless we started passing bad data to the handlers, which is unlikely.